io-engine/tests/lock_lba_range.rs

```diff
@@ -173,66 +110,42 @@ fn multiple_locks() {
     let reactor = Reactors::current();
 
     // First Lock
-    let (s, r) = unbounded::<()>();
-    let ctx1 = ShareableContext::new(1, 10);
-    let ctx_clone1 = ctx1.clone();
+    let (s, r) = unbounded();
     reactor.send_future(async move {
-        lock_range(
-            ctx_clone1.borrow_mut_ctx().deref_mut(),
-            ctx_clone1.borrow_ch().deref(),
-        )
-        .await
-        .unwrap();
-        s.send(()).unwrap();
+        let lock1 = lock_range(LbaRange::new(1, 10)).await.unwrap();
+        s.send(lock1).unwrap();
     });
-    reactor_poll!(r);
+    let lock1 = recv_from(r);
 
     // Second Lock
-    let (lock_sender, lock_receiver) = unbounded::<()>();
-    let ctx2 = ShareableContext::new(1, 5);
-    let ctx_clone2 = ctx2.clone();
+    let (lock_sender, lock_receiver) = unbounded();
     reactor.send_future(async move {
-        lock_range(
-            ctx_clone2.borrow_mut_ctx().deref_mut(),
-            ctx_clone2.borrow_ch().deref(),
-        )
-        .await
-        .unwrap();
-        lock_sender.send(()).unwrap();
+        let lock2 = lock_range(LbaRange::new(1, 5)).await.unwrap();
+        lock_sender.send(lock2).unwrap();
     });
     reactor_poll!(100);
 
     // First lock is held, second lock shouldn't succeed
     assert!(lock_receiver.try_recv().is_err());
 
     // First unlock
-    let (s, r) = unbounded::<()>();
+    let (s, r) = unbounded();
     reactor.send_future(async move {
-        unlock_range(
-            ctx1.borrow_mut_ctx().deref_mut(),
-            ctx1.borrow_ch().deref(),
-        )
-        .await
-        .unwrap();
+        unlock_range(lock1).await.unwrap();
         s.send(()).unwrap();
     });
     reactor_poll!(r);
 
     // Poll reactor to allow the second lock to be obtained
-    reactor_poll!(100);
+    let lock2 = recv_from(lock_receiver);
 
-    // First lock released, second lock should succeed.
-    assert!(lock_receiver.try_recv().is_ok());
+    // // First lock released, second lock should succeed.
+    // assert!(lock_receiver.try_recv().is_ok());
```

Comment:
does this need to be commented?

Reply:
I forgot to remove these lines. 
Now, lock lba range methods work a bit differently: lock_lba_range() returns a "lock" object which must be passed to unlock_lba_range(). This makes code safer because caller doesn't have to manually keep track on passing exactly the same context pointer and range to both lock() and unlock(). The "lock" object does it automatically.
Because of that, some tests were removed (the one testing for error handling when different args were passed to lock/unlock). Other tests were worked to use the new API.